src/test/store/profile-view.test.js

```diff
@@ -3604,3 +3604,27 @@ describe('getProcessedEventDelays', function() {
     ]);
   });
 });
+
+// This test is for 'mouseTimePosition' redux store
+describe('mouseTimePosition', function() {
+  function setup() {
+    const profile = getProfileFromTextSamples('A');
+
+    return storeWithProfile(profile.profile);
+  }
+
+  it('should get mouse time position', () => {
+    // Test just the redux store here.
+    const { dispatch, getState } = setup();
+
+    const mouseTimePosition = ProfileViewSelectors.getMouseTimePosition(
+      getState()
+    );
+
+    dispatch(ProfileView.changeMouseTimePosition(null));
+    expect(mouseTimePosition).toBeNull;
+
+    dispatch(ProfileView.changeMouseTimePosition(1000));
+    expect(mouseTimePosition).toBe(1000);
```

Comment:
This expect is returning undefined.

Reply:
The reason here, is that redux relies on immutable data, that is it's core strength. The selector needs to be run again after each action. I would recommend reading up on immutability here: https://redux.js.org/faq/immutable-data

Reply:
For instance:
```js

    dispatch(ProfileView.changeMouseTimePosition(null));
    expect(ProfileViewSelectors.getMouseTimePosition(
      getState()
    )).toBeNull();

    dispatch(ProfileView.changeMouseTimePosition(1000));
    expect(ProfileViewSelectors.getMouseTimePosition(
      getState()
    )).toBe(1000);
```